cardano-crypto-praos/cbits/private/ed25519_ref10.c

```diff
@@ -2775,12 +2759,12 @@ ge25519_elligator2(fe25519 x, fe25519 y, const fe25519 r, int *notsquare_p)
     fe25519_sq2(rr2, r);
     rr2[0]++;
     fe25519_invert(rr2, rr2);
-    fe25519_mul32(x, rr2, ed25519_A_32);
+    fe25519_scalar_product(x, rr2, ed25519_A_32);
```

Comment:
tl;dr: I'll use `fe25519_mul` for consistency, as it is what it is currently used in stable and in our libsodium fork. 

Good catch, I had trouble understanding what I did here (so I'll just explain it for future me, in case I come back to this). It's actually the same. The function `fe25519_mul32` is called `fe25519_scalar_product` in 1.0.18.  (https://github.com/jedisct1/libsodium/blob/4f5e89fa84ce1d178a6765b8b46f2b6f91216677/src/libsodium/include/sodium/private/ed25519_ref10_fe_25_5.h#L982). The difference between `fe25519_mul` and `fe25519_scalarproduct`/`fe25519_mul32`, is that the former multiplies two `fe25519` elements, while the second multiplies one `fe25519` times a `uint32_t`. Using the code from version 1.0.18, the function `fe25519_mul32` no longer exists, so I must have changed that by its replacement, `fe25519_scalar_product` (which I did [here](https://github.com/input-output-hk/libsodium/commit/8223276e69e7623ab1ea9e3d8c6fd67a440d7ed9)). 

The difference between `ed25519_A_32` and `curve25519_A` is that the [latter](https://github.com/input-output-hk/libsodium/commit/8223276e69e7623ab1ea9e3d8c6fd67a440d7ed9) is the `fe25519` representation of the [former](https://github.com/jedisct1/libsodium/blob/master/src/libsodium/crypto_core/ed25519/ref10/fe_25_5/constants.h#L23), which is a `uint32_t`. However, this is again a leftover from master/previous version, so I'll use the representation that version 1.0.28-stable has.